src/ingestion/sources/redshift.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+from .base import SQLAlchemyExtractor, SourceConfiguration, SQLAlchemySourceDialect, SQLAlchemySource
+from .postgresql import PostgreSQLSourceDialect
+
+class RedshiftSourceConfiguration(SourceConfiguration):
+    @classmethod
+    def validate(cls, configuration):
+        raise NotImplementedError
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "user": { "type": "string" },
+                "password": { "type": "string" },
+                "host": { "type": "string" },
+                "port": { "type": "string" },
+                "database": { "type": "string" },
+                "schema": { "type": "string" },
+            },
+            "secret": [ "password" ],
+        }
+
+    def connection_string(self) -> str:
+        configuration = json.loads(self.configuration)
+
+        return 'redshift+psycopg2://{user}:{password}@{host}:{port}/{database}'.format(
+            user=configuration.get('user'),
+            password=configuration.get('password'),
+            host=configuration.get('host'),
+            port=configuration.get('port'),
+            database=configuration.get('database'),
+            schema=configuration.get('schema'),
+        )
```

Comment:
Use sqlalchemy URI  function to create. Otherwise, you need to make sure you are not missing bugs like url encoding

you'd do better to super() and prepend redshift here as well.